### PR TITLE
fix(eslint-markdown): ignore exempt `img` tags in `require-image-title`

### LINKED ISSUE #727: Bug: ignore exempt `<img>` elements in `require-image-title`
### Environment

ESLint version: v9.39.5
@eslint/markdown version: v.8.0.3
eslint-markdown version: v0.16.0
Node version: v24.11.1
npm version: v11.6.4
Operating System: Windows 11

### Which language are you using?

commonmark & gfm

### What did you do?

<details>
<summary>Configuration</summary>

```js
import { defineConfig } from 'eslint/config';
import markdown from '@eslint/markdown';
import md from 'eslint-markdown';

export default defineConfig([
  {
    files: ['repro-commonmark.md'],
    plugins: {
      markdown,
      md,
    },
    language: 'markdown/commonmark',
    rules: {
      'md/require-image-title': 'error',
    },
  },
  {
    files: ['repro-gfm.md'],
    plugins: {
      markdown,
      md,
    },
    language: 'markdown/gfm',
    rules: {
      'md/require-image-title': 'error',
    },
  },
]);

```

</details>

```md
<img id="1" />
<img src="image.png" alt="" />
<img src="image.png" aria-hidden="true" />
```

### What did you expect to happen?

The rule should ignore the following elements.

- An `` element that has none of the `src`, `srcset`, or `alt` attributes.
- An `` element with `alt=""`.
- An `` element with `aria-hidden="true"`.

### What actually happened?

The rule reports that all three `` elements are missing a title attribute.

```
~/projects/stackblitz-starters-6yjgiguu 1s
❯ npx eslint repro-commonmark.md repro-gfm.md

/home/projects/stackblitz-starters-6yjgiguu/repro-commonmark.md
  1:1  error  Images should have a title attribute  md/require-image-title
  2:1  error  Images should have a title attribute  md/require-image-title
  3:1  error  Images should have a title attribute  md/require-image-title

/home/projects/stackblitz-starters-6yjgiguu/repro-gfm.md
  1:1  error  Images should have a title attribute  md/require-image-title
  2:1  error  Images should have a title attribute  md/require-image-title
  3:1  error  Images should have a title attribute  md/require-image-title

✖ 6 problems (6 errors, 0 warnings)
```

### Link to Minimal Reproducible Example

https://stackblitz.com/edit/stackblitz-starters-chv9dmic?file=eslint.config.mjs

### Participation

- [x] I am willing to submit a pull request for this issue.

### AI acknowledgment

- [x] No - I did not use AI to generate this issue report.
- [ ] Yes - minimal assistance (e.g., IDE autocomplete, small code completions, grammar fixes).
- [ ] Yes - substantive assistance (AI-generated non-trivial portions of code, tests, or documentation).

### Additional comments

This issue follows the discussion in #719 and tracks the `require-image-title` behavior separately, as requested.

### PR DESCRIPTION
**Pull request by electrohyun**
## Prerequisites checklist

- [x] I have read the [code of conduct](https://github.com/eslint-markdown/eslint-markdown/blob/main/CODE_OF_CONDUCT.md).
- [x] I have read the [contributing guidelines](https://github.com/eslint-markdown/eslint-markdown/blob/main/CONTRIBUTING.md).

## AI acknowledgment

Did you use AI tools to create any part of this pull request?

Please select one option and, if yes, briefly describe how AI was used (e.g., code, tests, docs) and which tool(s) you used.

- [x] No - I did not use AI for this contribution.
- [ ] Yes - minimal assistance (e.g., IDE autocomplete, small code completions, grammar fixes).
- [ ] Yes - substantive assistance (AI-generated non-trivial portions of code, tests, or documentation).
- [ ] Yes - other (please describe):

## What is the purpose of this pull request?

Updates `require-image-title` to ignore `` elements without any of the `src`, `srcset`, or `alt` attributes, as well as elements with `alt=""` or `aria-hidden="true"`.

## What changes did you make? (Give an overview)

Updated the `require-image-title` rule to ignore exempt `` elements.

Added tests and documentation for the new behaviors.

## How did you test this change?

I added the following test cases to cover the new behavior.

```
Valid:
- An `<img>` element without `src`, `srcset`, or `alt` should be ignored
- An `<img>` element with `alt=""` should be ignored
- An `<img>` element with `aria-hidden="true"` should be ignored

Invalid:
- An `<img>` element with an empty `src` and no title should be reported
- An `<img>` element with only `srcset` and no title should be reported
- An `<img>` element with a non-empty `alt` and no title should be reported
- An `<img>` element with a lone `aria-hidden` attribute should be reported
- An `<img>` element with `aria-hidden="false"` and no title should be reported
- An image without a title following ignored images should still be reported
```

## Related Issues

fixes #727

## Is there anything you'd like reviewers to focus on?

I followed the same structure as the merged #720. Please let me know if you have any suggestions for improvement.

### CODE REVIEW

File: packages/eslint-markdown/src/rules/require-image-title.test.ts [hunk 1]
```diff
@@ -308,6 +320,90 @@ ruleTester('require-image-title', rule, {
...
+          line: 1,
+          column: 1,
+          endLine: 1,
+          endColumn: 45,
+        },
+      ],
+    },
+    {
+      name: 'Html node with non-empty alt attribute and without title attribute',
+      code: '<img alt="Alt text">',
```
Comment (lumirlumir) [inline packages/eslint-markdown/src/rules/require-image-title.test.ts:363]:
```js
      code: '<img alt>',
```

```js
      code: '<img alt="">',
```

Could we add two more test cases where `alt` with no assigned value or empty?

File: packages/eslint-markdown/src/rules/require-image-title.test.ts [hunk 2]
```diff
@@ -72,6 +72,18 @@ ruleTester('require-image-title', rule, {
       name: 'Html node with title attribute',
       code: '<img src="https://example.com/image.jpg" title="title">',
     },
+    {
+      name: 'Html node without src, srcset, or alt attribute',
+      code: '<img id="1">',
```
Comment (lumirlumir) [inline packages/eslint-markdown/src/rules/require-image-title.test.ts]: Could we add one more following test case where `img` has no additional attributes?

```js
      code: '<img>',
```

**Review by lumirlumir — COMMENTED** [review]
Thanks for the PR!

I've left a few more suggestions regarding the test cases. Otherwise, LGTM.

**Review by lumirlumir — APPROVED** [review]
Thanks for the contribution!
